programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -256,6 +275,153 @@ static int start_event_socket(uv_loop_t *l) {
     return -1;
 }
 
+static void tnl_transfer_rates_cb(const tunnel_identity_metrics *metrics, void *ctx) {
+    tunnel_identity *tnl_id = ctx;
+    tnl_id->Metrics = calloc(1, sizeof(struct tunnel_metrics_s));
+    tnl_id->Metrics->Up = (int) strtol(metrics->up, NULL, 10);
+    tnl_id->Metrics->Down = (int) strtol(metrics->down, NULL, 10);
+}
+
+
+static char* addUnit(int count, char* unit) {
+    char* result = malloc(MAXMESSAGELEN);
+
+    if ((count == 1) || (count == 0)) {
+        snprintf(result, MAXMESSAGELEN, "%d %s", count, unit);
+    } else {
+        snprintf(result, MAXMESSAGELEN, "%d %ss", count, unit);
+    }
+    return result;
+}
+
+static string convert_seconds_to_readable_format(int input) {
+    int seconds = input % (60 * 60 * 24);
+    int hours = ((double) seconds / 60 / 60);
+    seconds = input % (60 * 60);
+    int minutes = ((double) seconds)/ 60;
+    seconds = input % 60;
+    char* result = malloc(MAXMESSAGELEN);
+
+    if (hours > 0) {
+        snprintf(result, MAXMESSAGELEN, "%s %s %s", addUnit(hours, "hour"), addUnit(minutes, "minute"), addUnit(seconds, "second"));
```

Comment:
this leaks memory: `addUnit` allocates string

Reply:
freed the memory